src/recordlinker/tuning/prob_calc.py

```diff
@@ -98,36 +97,40 @@ def calculate_and_sort_tuning_scores(
     :returns: A dictonary mapping the names of the algorithm's passes to
       a tuple containing sorted lists of class RMS scores.
     """
-    context: ag.AlgorithmContext = algorithm.algorithm_context
-    sorted_scores = {}
-
-    # Tuning generates results per pass, so we'll generate one result set
-    # per pass of the given algorithm
-    for idx, algorithm_pass in enumerate(algorithm.passes):
-        max_points_in_pass: float = sum(
-            [log_odds[str(e.feature)] or 0.0 for e in algorithm_pass.evaluators]
-        )
-        true_match_scores: list[float] = _score_pairs_in_class(
-            true_match_pairs, log_odds, max_points_in_pass, algorithm_pass, context
-        )
-        non_match_scores: list[float] = _score_pairs_in_class(
-            non_match_pairs, log_odds, max_points_in_pass, algorithm_pass, context
-        )
-        true_match_scores = sorted(true_match_scores)
-        non_match_scores = sorted(non_match_scores)
-        sorted_scores[
-            algorithm_pass.label or f"pass_{idx}"
-        ] = (true_match_scores, non_match_scores)
+    sorted_scores: dict[str, typing.Tuple[list[float], list[float]]] = {
+        p.resolved_label: ([], []) for p in algorithm.passes
+    }
+    max_points: dict[str, float] = {
+        p.resolved_label: sum([log_odds.get(e.feature, 0.0) for e in p.evaluators])
+        for p in algorithm.passes
+    }
+
+    # Both true-match and non-match pairs are iterables and can only be accessed once,
+    # we need to use the pairs to calculate values for all passes in the Algorithm, thus we
+    # need to iterate over the pairs first, then the passes
+    for pair in true_match_pairs:
+        for key, score in _score_records_in_pair(pair, log_odds, max_points, algorithm).items():
+            if score > 0.0:
+                # ignore true-matches with no agreement
+                sorted_scores[key][0].append(score)
```

Comment:
Nit: Just thinking about possible downstream modifications, we may want to leave this in `estimate_rms_bounds` instead of pre-filtering here. We know there's the possibility that NBS adds some more analysis around tuning at some point (based on Julie's asks to separate log-odds calculation from RMS calculation), and if as part of that they want to look at the scores for the true-match class it would probably be useful to have the 0's included so they could analyze the distribution. Taking the 0s out only really matters for the RMS bounds so my vote would be to leave it there. Non-blocker though!

Reply:
ok.  Just thought it might be a little optimization to remove them early if we just discard later.  I can revert though.